client/network/src/protocol.rs

```diff
@@ -632,7 +692,7 @@ impl<B: BlockT, H: ExHashT> Protocol<B, H> {
 					"Received no longer supported legacy request from {:?}",
 					who
 				);
-				self.disconnect_peer(&who);
+				self.behaviour.disconnect_peer(&who, sc_peerset::SetId::from(0));
```

Comment:
Just by reading `sc_peerset::SetId::from(0)` it is hard to know that one is disconnecting from the block request peer set. What do you think of defining a constant e.g. `BLOCK_ANNOUNCE_PEER_SET` which is used everywhere instead of `sc_peerset::SetId::from(0)`. That would help me understand the context.